### ISSUE TITLE
Null Socket (SMTP CONNECTION)

### ISSUE THREAD

**Issue by jquequezana** [zone-eu/zone-mta#138]
```
353 info SMTP/feeder/2494 SMTP feeder MTA server started
    354 /home/reliam/utils/zonemta/zone-mta-template_/node_modules/smtp-server/lib/smtp-connection.js:71
    355         this.remoteAddress = (options.remoteAddress || this._socket.remoteAddress || '').replace(/^::ffff:/, '');
    356                                                                     ^
    357 
    358 TypeError: Cannot read property 'remoteAddress' of undefined
    359     at new SMTPConnection (/home/reliam/utils/zonemta/zone-mta-template_/node_modules/smtp-server/lib/smtp-connection.js:71:69)
    360     at SMTPServer.connect (/home/reliam/utils/zonemta/zone-mta-template_/node_modules/smtp-server/lib/smtp-server.js:103:26)
    361     at process.on (/home/reliam/utils/zonemta/zone-mta-template_/node_modules/zone-mta/services/receiver.js:148:27)
    362     at process.emit (events.js:182:13)
    363     at emit (internal/child_process.js:797:12)
    364     at process._tickCallback (internal/process/next_tick.js:174:19)
    365 ERR! Process Uncaught exception
    366 ERR! Process Error [ERR_IPC_CHANNEL_CLOSED]: Channel closed
    367 ERR! Process     at ChildProcess.target.send (internal/child_process.js:613:16)
    368 ERR! Process     at SMTPProxy.connection (/home/reliam/utils/zonemta/zone-mta-template_/node_modules/zone-mta/lib/receiver/smtp-proxy.js:131:15)
    369 ERR! Process     at Server.server.net.createServer.socket (/home/reliam/utils/zonemta/zone-mta-template_/node_modules/zone-mta/lib/receiver/smtp-proxy.js:178:59)
    370 ERR! Process     at Server.emit (events.js:182:13)
    371 ERR! Process     at Server.EventEmitter.emit (domain.js:442:20)
    372 ERR! Process     at TCP.onconnection (net.js:1550:8)
    373 ERR! Process  { Error [ERR_IPC_CHANNEL_CLOSED]: Channel closed
    374 ERR! Process     at ChildProcess.target.send (internal/child_process.js:613:16)
    375 ERR! Process     at SMTPProxy.connection (/home/reliam/utils/zonemta/zone-mta-template_/node_modules/zone-mta/lib/receiver/smtp-proxy.js:131:15)
    376 ERR! Process     at Server.server.net.createServer.socket (/home/reliam/utils/zonemta/zone-mta-template_/node_modules/zone-mta/lib/receiver/smtp-proxy.js:178:59)
    377 ERR! Process     at Server.emit (events.js:182:13)
    378 ERR! Process     at Server.EventEmitter.emit (domain.js:442:20)
    379 ERR! Process     at TCP.onconnection (net.js:1550:8)
    380 ERR! Process   stack: 'Error [ERR_IPC_CHANNEL_CLOSED]: Channel closed\n    at ChildProcess.target.send (internal/child_process.js:613:16)\n    at SMTPProxy.connection (/home/reliam/utils/zonemta/zone-mta-template_/node_modules/z        one-mta/lib/receiver/smtp-proxy.js:131:15)\n    at Server.server.net.createServer.socket (/home/reliam/utils/zonemta/zone-mta-template_/node_modules/zone-mta/lib/receiver/smtp-proxy.js:178:59)\n    at Server.emit (events.js:182:        13)\n    at Server.EventEmitter.emit (domain.js:442:20)\n    at TCP.onconnection (net.js:1550:8)' }
    381 info Process Server closing down...
```

my temporary fix
  
```
file: /zone-mta/services/receiver.js
128 process.on('message', (m, socket) => {
129     if (m === 'socket') {
130         if (typeof socket == 'undefined') {
131             log.error('SMTP/' + currentInterface + '/' + process.pid, 'Null Socket');
132             return;
133         }
```

**Comment by andris9** [comment]
Did you get a lot of these undefined sockets or was this a one time error? I wonder how it ended up in that stage

**Comment by jquequezana** [comment]
One time error, and the feeder die ... need reset the zone after that ... I see the error two times after sent around 500k emails.
With my mod I can see many times the null socket but after all the zone start sending again 

```
  324 info Sender/myip287/4710 [0b131597119c9ce79330] Starting sending for myip287
  325 info Sender/myip314/4875 [f96f7cbde2094a287e4a] Starting sending for myip314
  326 ERR! SMTP/feeder/5006 Null Socket
  327 ERR! SMTP/feeder/5006 Null Socket
  328 ERR! SMTP/feeder/5006 Null Socket
  329 ERR! SMTP/feeder/5006 Null Socket
  330 ERR! SMTP/feeder/5006 Null Socket
  331 ERR! SMTP/feeder/5006 Null Socket
  332 ERR! SMTP/feeder/5006 Null Socket
  333 ERR! SMTP/feeder/5006 Null Socket
  334 ERR! SMTP/feeder/5006 Null Socket
  335 ERR! SMTP/feeder/5006 Null Socket
  336 ERR! SMTP/feeder/5006 Null Socket
....
28694 ERR! SMTP/feeder/5006 Null Socket
28695 ERR! SMTP/feeder/5006 Null Socket
28696 ERR! SMTP/feeder/5006 Null Socket
28697 info Sender/myip325/4942 [c84e08322013db608249] Starting sending for myip325
28698 info SMTP/feeder/5006 Connection from localhost.localdomain[127.0.0.1]:47666
28699 info SMTP/feeder/5006 Connection from localhost.localdomain[127.0.0.1]:47668
28700 info SMTP/feeder/5006 Connection from localhost.localdomain[127.0.0.1]:47670
28701 info SMTP/feeder/5006 Connection from localhost.localdomain[127.0.0.1]:47672
28702 info SMTP/feeder/5006 Connection from localhost.localdomain[127.0.0.1]:47674
28703 info SMTP/feeder/5006 Connection from localhost.localdomain[127.0.0.1]:47676
28704 info SMTP/feeder/5006 Connection from localhost.localdomain[127.0.0.1]:47678
28705 info SMTP/feeder/5006 Connection from localhost.localdomain[127.0.0.1]:47680
```

Need say that I had and forever sender 
> # forever list
> data:        uid  command             script            forever pid   id logfile                 uptime         
> data:    [0] elcP /usr/local/bin/node sender.js         31134   32582    /root/.forever/elcP.log 0:2:21:24.319

**Comment by andris9** [comment]
This should be fixed in v1.7.8 - at least it should now throw anymore, no idea what happens to these ghost sockets though.

could you also try to start your 500 zones with the new version? i changed the spawning code to wait for a message from the child process that it actually has started before moving on to the next zone.
